### PR TITLE
#738 commandlet for lazydocker added

### LINKED ISSUE #738: CLI Lazy Docker integration
As a devon IDE user, I want to have Lazy Docker available in the devon IDE for my docker installation.

https://github.com/jesseduffield/lazydocker
 
Acceptance criteria:

- [x] Mirror is added.
- [x] There is a new commandlet that allows to install and launch Lazydocker from the CLI
- [x] Installation of the [binary releases](https://github.com/jesseduffield/lazydocker#binary-release-linuxosxwindows)
- [x] During the installation process I get information about the current status of the installation
- [x] Lazydocker can be installed and used together with any of the devon IDE standard tools without negative interference
- [x] Check Docker requirements (Docker >= 1.13 (API >= 1.25); Docker-Compose >= 1.23.2 (optional))
- [x] The new commandlet is documented on the website
- [x] Add licence documentation

### PR DESCRIPTION
**Pull request by arexhepi7**
Fixed #738

### CODE REVIEW

**Comment by hohwille** [comment]
Thanks for the PR.
Can you check with [DoD](https://github.com/devonfw/ide/blob/master/documentation/DoD.asciidoc)?
```
Running test #2: test-shellcheck

In scripts/command/lazydocker line 20:
  cd ${LAZYDOCKER_HOME}
  ^-------------------^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
     ^----------------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  cd "${LAZYDOCKER_HOME}" || exit

For more information:
  https://www.shellcheck.net/wiki/SC2164 -- Use 'cd ... || exit' or 'cd ... |...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
Error:  Failed running test #2: test-shellcheck - exit code 1
```

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:27]: Seems that lazydocker has some issues to run on windows seamless in any shell.
Actually my vision would be that you could just setup lazydocker and then do not need `devon` command and just run `lazydocker` in any shell on any os.
Is that vision out of reach?
Can you share what errors or problems arise when you try this?

Comment (cinnamon-coder-hub) [reply scripts/src/main/resources/scripts/command/lazydocker:27]: This comments issue will be addressed in #842 as recently discussed and the comment can therefore be closed.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:20]: should actually be on `PATH` so `lazydocker` should work directly (instead of `./lazydocker`).
May only be required to start via powershell.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:53]: `docker version --format '{{.Client.Version}}'`

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:55]: `docker version --format '{{.Client.APIVersion}}'`

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:61]: So this WARNING is printed every time regardless if the versions are fine or not?

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:74]: so `4`  means that both versions are higher and `0` that both versions are identical. So `2` means one is equal and one is higher - because you are in the else so it can not be the `1+1` case? Somehow seems correct but does this `ressum` really make anything better? IMHO it is only causing confusion and making the code hard to read.

Why not just kick out `ressum` and do `else` here?
The only possible results are 0 (equal), 1 (greater), 2 (lower):
https://github.com/devonfw/ide/blob/3c49143d7a3167378f90a25daf09b1d890d2dd27/scripts/src/main/resources/scripts/functions#L1381

To avoid confusion it would even be helpful to switch the arguments:
```
doVersionCompare "${DOCKERENGVERSION}" "1.3.0"
local res1=$?
```

so `1` means greater (${DOCKERENGVERSION} > 1.3.0) and `2` means lower (${DOCKERENGVERSION} < 1.3.0).

So these are the cases (without switching version arguments):
|res1|res2|result|
|----|----|------|
|  0  | 0  | fine, both versions equal |
|  0 |  1  | bad, ApiVersion is lower |
|  1 |  0  | bad, Version is lower |
|  1 |  1  | bad, both versions lower |
|  0 |  2  | fine, version equal, ApiVersion higher |
|  1 |  2  | bad, version lower, ApiVersion higher |
|  ... |  ... | ... |

So to make it short, the first if is covering all bad cases, and `else` would be fine.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:79]: this case can never happen and could therefore be dropped.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:30]: you should add
```
doDevonCommand docker setup silent
```
as before what will ensure that docker/rancher is first installed if not already present.

**Review by hohwille — CHANGES_REQUESTED** [review]
@alfeilex and antigoni: Thanks for this PR. Almost prefect 👍 
I added some review comments.
Can you check and rework so that we can merge?

**Comment by alfeilex** [comment]
> @alfeilex and antigoni: Thanks for this PR. Almost prefect 👍 I added some review comments. Can you check and rework so that we can merge?

I have been mentioned here incorrectly. @cinnamon-coder-hub has created the PR with @arexhepi7  .

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (cinnamon-coder-hub) [reply scripts/src/main/resources/scripts/command/lazydocker]: Now there is a separate if condition for the case that docker-compose is at an insufficient version or not running. So users with the propper version won't be bothered by this warning anymore.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:59]: this is outputting nothing but the version. So what is the need of the `sed` command?

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:61]: Same here, can you remove the `sed` pipe?

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:27]: This seems to be a missunderstanding.
If `doSetup` was called then `lazydocker` has to be installed and on the `PATH`. Otherwise this is a bug of `doSetup`.
IMHO we do not need this block.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:30]: so what I actually wanted to say is the you do not need to do `cd ${LAZYDOCKER_HOME}` and can just call `lazydocker` without the `./` what also makes the assumption that it has to be directly in `software/lazydocker` and we get that content from an external source - maybe in future releases they decide to move the binary to a `bin` subfolder.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/lazydocker:109]: if we want to have this feature in the future for all commandlets, we could simply add it to `functions` so it does not have to be implemented into every commandlet. However, this is just a side-nore for the record. We can file a separate issue for this. Let us get this lazydocker completed and merged.

**Review by hohwille — CHANGES_REQUESTED** [review]
@alfeilex thanks for the rework. The version compatibility checks are nicely improved, etc. 👍 

Still I have some last remarks. I would have done that as constructive review by directly changing the code but as I am going on vacation and do not want to interfere or break anything due to lack of time for testing last minute changes, I leave this up to you for the last round.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (hohwille) [reply scripts/src/main/resources/scripts/command/lazydocker:109]: I created issue #847 for this.

File: scripts/src/main/resources/scripts/command/lazydocker
Comment (cinnamon-coder-hub) [reply scripts/src/main/resources/scripts/command/lazydocker:29]: The lazydocker cmdlet should work, but this additional command seems to make problems in my case. For Antigoni it is fine again. I suspect that this is the case because of some changes that I made for testing purposes, but I will look deeper into this next week and make another commit if necessary.

**Review by hohwille — APPROVED** [review]
Thanks for the final rework. Now everything is just perfect 👍

**Pull request merged**
